src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerConversionExtensions.cs

```diff
@@ -43,7 +43,7 @@ public static class JaegerConversionExtensions
         private const long TicksPerMicrosecond = TimeSpan.TicksPerMillisecond / 1000;
         private const long UnixEpochMicroseconds = UnixEpochTicks / TicksPerMicrosecond; // 62,135,596,800,000,000
 
-        public static JaegerSpan ToJaegerSpan(this SpanData span)
+        public static void ToJaegerSpan(this SpanData span, out JaegerSpan jaegerSpan)
```

Comment:
Why change the signature to use out param instead of returning JaegerSpan?

Reply:
Just one of the things I tried (of many) to get the allocations down. I'll put it back, just tested and this had no impact on the perf. I made some things structs which were only being used while the logic was executing to build up the buffer, I thought that would reduce allocations but the memory actually jumped up! After a lot of messing around, it seemed to be this pattern:
```csharp
if (this.References is IEnumerable<JaegerSpanRef> references)
```
Seems like that "is" operation does a boxing and actually copies the data from the stack to the heap? That was a first to see, to be honest, and I'm not totally sure what was going on. When I switched them to "!= null" checks, memory started behaving as I was expecting. I haven't had a chance to dig more into it.